src/checkers/inference/InferenceVisitor.java

```diff
@@ -208,7 +208,7 @@ public void mainIs(AnnotatedTypeMirror ty, AnnotationMirror mod, String msgkey,
         annoIs(ty, main, mod, msgkey, node);
     }
 
-    public void mainIsSubtype(AnnotatedTypeMirror ty, AnnotationMirror mod, String msgkey, Tree node) {
+    public boolean mainIsSubtype(AnnotatedTypeMirror ty, AnnotationMirror mod, String msgkey, Tree node) {
```

Comment:
If I remember our discussions, I don't think you want to change the return type.
The else branch should still issue the error message if the subtype relation doesn't hold.
You didn't adapt any callers, so doing this will ignore these constraints now!

Reply:
Got it. But without changing it to `boolean` return type, PICO can't override this method. Maybe I will talk to Piyush to see if he met this problem before

Reply:
Wasn't the only reason PICO needed to override this method to fix this problem?
So, if we fix the problem here, PICO shouldn't have to override the method at all.

Reply:
Some other methods also rely on the boolean return value from `mainIsSubtype`.

`mainIsSubtype` is called by `isAdaptedSubtype`, and `isAdaptedSubtype` is called in the `isValidUse` in `PICOInferenceVisitor.java`. And basically `isValidUse` method will determine if the qualifier on use type is valid or not based on the return value from `mainIsSubtype`. Maybe we can think of a nicer way to refactor the these methods.